demos/nrf52-beacon/src/main.rs

```diff
@@ -81,19 +61,25 @@ const APP: () = {
         )
         .unwrap();
 
-        init::LateResources {
-            radio,
-            beacon,
-            beacon_timer: ctx.device.TIMER1,
-        }
+        ctx.spawn.update().ok();
+
+        init::LateResources { radio, beacon }
     }
 
     /// Fire the beacon.
-    #[task(binds = TIMER1, resources = [beacon_timer, beacon, radio])]
-    fn timer1(ctx: timer1::Context) {
-        // Acknowledge event so that the interrupt doesn't keep firing
-        ctx.resources.beacon_timer.events_compare[0].reset();
-
+    #[task(schedule = [update], resources = [beacon, radio])]
+    fn update(ctx: update::Context) {
         ctx.resources.beacon.broadcast(ctx.resources.radio);
+
+        ctx.schedule
+            .update(ctx.scheduled + 1_066_666.cycles())
```

Comment:
Can you add a comment explaining where the 1066666 cycles number comes from?

Reply:
sure

Reply:
good catch, it was wrong by factor 10 anyways :facepalm: